### PR TITLE
Fix platform availability of some Unix constants

### PR DESCRIPTION
**Pull request by AlexWaygood**

### CODE REVIEW

File: stdlib/os/__init__.pyi [hunk 1]
```diff
@@ -70,9 +70,20 @@ if sys.platform != "win32":
...
+        # sys.platform could have one of many values on a FreeBSD Python build,
+        # so the sys-module docs recommend doing `if sys.platform.startswith('freebsd')`
+        # to detect FreeBSD builds. Unfortunately that would be too dynamic
+        # for type checkers, however.
+        SF_NODISKIO: int
+        SF_MNOWAIT: int
+        SF_SYNC: int
+
+        if sys.version_info >= (3, 11):
+            SF_NOCACHE: int
```
Comment (AlexWaygood) [inline stdlib/os/__init__.pyi:86]: `os` docs:

- https://docs.python.org/3/library/os.html#os.SF_NODISKIO
- https://docs.python.org/3/library/os.html#os.SF_NOCACHE

Source code: https://github.com/python/cpython/blob/de2a4036cbfd5e41a5bdd2b81122b7765729af83/Modules/posixmodule.c#L16562-L16575

File: stdlib/select.pyi [hunk 2]
```diff
@@ -137,7 +137,6 @@ if sys.platform == "linux":
     EPOLLRDNORM: int
     EPOLLWRBAND: int
     EPOLLWRNORM: int
-    EPOLL_RDHUP: int
```
Comment (AlexWaygood) [inline stdlib/select.pyi:140]: I don't think this ever existed at runtime. I think it was just a figment of our imagination. Note the name is very similar to `select.EPOLLRDHUP`, which _does_ exist at runtime (and which we already have elsewhere in the stub).

File: stdlib/_curses.pyi [hunk 3]
```diff
@@ -61,7 +61,8 @@ if sys.platform != "win32":
     A_DIM: int
     A_HORIZONTAL: int
     A_INVIS: int
-    A_ITALIC: int
+    if sys.platform != "darwin":
+        A_ITALIC: int
```
Comment (AlexWaygood) [inline stdlib/_curses.pyi:65]: Docs here: https://docs.python.org/3/library/curses.html#curses.A_ITALIC

Immediately above this table, the docs state:

> Some constants are available to specify character cell attributes. The exact constants available are system dependent.

Comment (JelleZijlstra) [reply stdlib/_curses.pyi:65]: I tried and it indeed exists on Linux but not macOS

File: stdlib/os/__init__.pyi [hunk 4]
```diff
@@ -70,9 +70,20 @@ if sys.platform != "win32":
         POSIX_FADV_WILLNEED: int
         POSIX_FADV_DONTNEED: int
 
-    SF_NODISKIO: int
-    SF_MNOWAIT: int
-    SF_SYNC: int
+    if sys.platform != "linux" and sys.platform != "darwin":
+        # In the os-module docs, these are marked as being available
+        # on "Unix, not Emscripten, not WASI."
```
Comment (JelleZijlstra) [inline stdlib/os/__init__.pyi:75]: So maybe we should change the os docs?

Can confirm no os.SF_* contants on Linux or macOS.

File: stdlib/signal.pyi [hunk 5]
```diff
@@ -47,6 +45,9 @@ class Signals(IntEnum):
         SIGWINCH: int
         SIGXCPU: int
         SIGXFSZ: int
+        if sys.platform != "linux":
```
Comment (JelleZijlstra) [inline stdlib/signal.pyi:48]: For me these indeed exist on macOS but not Linux.

**Review by JelleZijlstra — APPROVED** [review]
Thanks!

File: stdlib/os/__init__.pyi (hunk 4 above)
Comment (AlexWaygood) [reply stdlib/os/__init__.pyi:75]: > So maybe we should change the os docs?

Yeah, I think so

**Pull request merged**